src/main/java/com/treasuredata/client/TDHttpClient.java

```diff
@@ -371,7 +372,7 @@ public RequestContext withRootCause(TDClientException e)
     }
 
     protected <Result> Result submitRequest(RequestContext context, TDHttpRequestHandler<Result> handler)
```

Comment:
Better to create a branch called PTD-XXX, and have PTD-XXX in the title as well as commit message. This is for tracking purpose.

Reply:
👍  I wasn't sure if I needed to do this since it's a public repo.

Reply:
Oh... I see. From other comments and PRs, better not to do it 🙇

Reply:
I just noticed that this a public repo, meaning the usage is very wide. I start to feel it is better to use a workaround so that this PR is not needed. For example, we can look into TDClientException, and then we can know what exception is wrapped. To be honest, it is difficult for me to decide the affect range of this PR, and it is why I think better to avoid changing this repo if possible.

Reply:
From a customer perspective, at the moment, if a thread is interrupted, he might have to handle 
- TDClientException (with ErrorType == INTERRUPTED)
- InterruptedIOException

However, the public signature of the methods specifically mentions that only TDClientExceptions will be thrown:
```
    public List<TDTable> listTables(String databaseName)
            throws TDClientException
```

So in a sense, the public API contract is not respected at the moment. It would be with this fix.
The only way for the customer to catch InterruptedIOException is to catch all exceptions as the compiler would prevent him from writing:
```
try {

} 
catch (TDClientException e) {

}
catch (InterruptedIOException e) {

}
```

Reply:
It might not be perfect for clients to handle interruptions, but still better not change the specification unless we are very confident and have enough reason. We should be very careful when changing public contract.